src/duplicacy_snapshotmanager.go

```diff
@@ -2264,12 +2264,21 @@ func (manager *SnapshotManager) CheckSnapshot(snapshot *Snapshot) (err error) {
 		lastEntry = entry
 	}
 
+	lastEntry = nil
 	for _, entry := range entries {
 
 		if !entry.IsFile() || entry.Size == 0 {
 			continue
 		}
 
+		if lastEntry != nil && 
+		   lastEntry.StartChunk == entry.StartChunk && lastEntry.EndChunk == entry.EndChunk && 
+		   lastEntry.StartOffset == entry.StartOffset && lastEntry.EndOffset == entry.EndOffset {
+			// another entry sharing the same data
```

Comment:
Not necessarily that uncommon. I ran into it quite often, with something producing multiple copies of files with identical content (although in my case they were quite small). 

I apologize if any constraints are broken by the small file deduplication changes. I do not know all the intricacies of your implementation, so please fix anything that need fixing!

Reply:
In support of @kairisku, it appears that the main branch already does something similar. Details:
 - (a) since entries are sorted by chunk index, any two identical small files in this PR will be adjacent and would incorrectly trigger an error later in this loop, i.e. ` entry.StartOffset < lastOffset`. In fact:
 - (b) Adjacent overlapping entries occur in @gilbertchen's main branch as well, **when using fixed chunk sizes**. (Each small file then ends up in a separate chunk, so identical small files will have entries with identical Start/EndOffset indexes.) Gilbert avoids the extraneous error by surrounding that test with  `if entry.StartOffset > 0 {...}` which coincidentally excludes all small files  when using fixed chunk sizes. Kai's test is simply a more general version of the same exclusion.  [Incidentally, I don't understand why the if clause is there, otherwise.]
 - (c) @gilbertchen could you clarify what you mean by "Some constraints below will be broken by the small file deduplication changes."?

Reply:
`entry.StartOffset < lastOffset` you mentioned in (a) is one of such constraints.  I'm not sure if there are more.

Reply:
Thanks for the reply, @gilbertchen. I think I understand your original complaint -- that we should simply make new elements in the `chunkHashes` and `chunkLengths` arrays rather than have two `entries` point to the same location.
I can think of a way of fixing it if absolutely necessary. But is anything broken this way?  Does the program _need_ `entry.StartOffset < lastOffset` to function or is that "constraint" simply an overly conservative rule in the function `CheckSnapshot`? (i.e. where will the program break if two `entries` share the same `chunkHashes` locations? Perhaps the correct constraint should be `entry.EndOffset < lastOffset` for example) Anyway, as I started: if it absolutely must be fixed I could probably do something, but I'd be very interested in your response.